### PR TITLE
[Driver][DragonFly] Fixes for linker path and command-line option handling

### PR DESCRIPTION
**Pull request by brad0**
- Add in some other linker command line options that the other BSD's handle
- Make use of AddFilePathLibArgs()
- Handle OpenMP

### CODE REVIEW

File: clang/lib/Driver/ToolChains/DragonFly.cpp [hunk 1]
```diff
@@ -115,16 +117,15 @@ void dragonfly::Linker::ConstructJob(Compilation &C, const JobAction &JA,
           Args.MakeArgString(getToolChain().GetFilePath("crtbegin.o")));
   }
 
-  Args.addAllArgs(CmdArgs, {options::OPT_L, options::OPT_T_Group});
+  Args.addAllArgs(CmdArgs,
```
Comment (MaskRay) [inline clang/lib/Driver/ToolChains/DragonFly.cpp:120]: Drop -Z: ld64 specific linker option that is not in GNU ld or ld.lld

     -Z          Do not search the standard directories when searching for
                 libraries and frameworks.

File: clang/test/Driver/dragonfly.c [hunk 2]
```diff
@@ -26,3 +27,14 @@
 // RELOCATABLE-NOT: "-dynamic-linker"
 // RELOCATABLE-NOT: "-l
 // RELOCATABLE-NOT: {{.*}}crt{{[^./]+}}.o
+
+// Check that the new linker flags are passed to DraonFly
+// RUN: %clang --target=x86_64-unknown-dragonfly -s -### %s 2>&1 \
```
Comment (MaskRay) [inline clang/test/Driver/dragonfly.c:32]: Test all driver linker options in one %clang command.

**Comment by brad0** [comment]
Fix a typo in a comment.

File: clang/lib/Driver/ToolChains/DragonFly.cpp [hunk 3]
```diff
@@ -56,7 +56,9 @@ void dragonfly::Linker::ConstructJob(Compilation &C, const JobAction &JA,
                                      const InputInfoList &Inputs,
                                      const ArgList &Args,
                                      const char *LinkingOutput) const {
-  const Driver &D = getToolChain().getDriver();
+  const toolchains::DragonFly &ToolChain =
```
Comment (MaskRay) [inline clang/lib/Driver/ToolChains/DragonFly.cpp:59]: `const auto &` as `static_cast` makes the type clear

**Review by MaskRay — APPROVED** [review]

**Comment by AaronBallman** [comment]
This appears to have broken some bots:
https://lab.llvm.org/buildbot/#/builders/119/builds/15633
https://lab.llvm.org/buildbot/#/builders/60/builds/14449

Can you investigate and get those bots back to green?

**Comment by AaronBallman** [comment]
> This appears to have broken some bots: https://lab.llvm.org/buildbot/#/builders/119/builds/15633 https://lab.llvm.org/buildbot/#/builders/60/builds/14449
> 
> Can you investigate and get those bots back to green?

Ping?

**Comment by MaskRay** [comment]
> > This appears to have broken some bots: [lab.llvm.org/buildbot/#/builders/119/builds/15633](https://lab.llvm.org/buildbot/#/builders/119/builds/15633) [lab.llvm.org/buildbot/#/builders/60/builds/14449](https://lab.llvm.org/buildbot/#/builders/60/builds/14449)
> > Can you investigate and get those bots back to green?
> 
> Ping?

fixed by e65721c3a1ee3ee4aa43499882f6d71dc9368daf

I guess this Windows bot is related to the CMake `DEFAULT_SYSROOT` variable that I want to deprecate in https://reviews.llvm.org/D158218

**Comment by AaronBallman** [comment]
> > > This appears to have broken some bots: [lab.llvm.org/buildbot/#/builders/119/builds/15633](https://lab.llvm.org/buildbot/#/builders/119/builds/15633) [lab.llvm.org/buildbot/#/builders/60/builds/14449](https://lab.llvm.org/buildbot/#/builders/60/builds/14449)
> > > Can you investigate and get those bots back to green?
> > 
> > 
> > Ping?
> 
> fixed by [e65721c](https://github.com/llvm/llvm-project/commit/e65721c3a1ee3ee4aa43499882f6d71dc9368daf)
> 
> I guess this Windows bot is related to the CMake `DEFAULT_SYSROOT` variable that I want to deprecate in https://reviews.llvm.org/D158218

Thank you for the fix! I'll keep my eyes on the bots to make sure they go back to green and report back if they don't. :-)

**Pull request merged**